### PR TITLE
update datamodel

### LINKED ISSUE #343: broken CI - numpy 2.0.0?
## Describe the bug
The current tests are broken. 

Screenshot 2024-07-01 at 13 55 02

https://github.com/openclimatefix/uk-pv-national-gsp-api/actions/runs/9744126472/job/26889085899

## Expected behavior
Numpy 2.0.0 was release

## Additional context
We could pin back, or try to make the code work for numpy 2.0.0

### PR DESCRIPTION
**Pull request by mduffin95**
# Pull Request

## Description

- Update `nowcasting_datamodel` to the latest version

Fixes #343 

## How Has This Been Tested?

Ran existing unit tests.

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by peterdudfield** [comment]
This ready to go @mduffin95 , thanks for looking at this?

**Comment by mduffin95** [comment]
@peterdudfield It's not 100% there yet. I'm getting the following in tests:

>  AttributeError: 'str' object has no attribute 'tzinfo'

It is occurring in the field_validator methods for fields of type `datetime` in `forecast.py` within `nowcasting_datamodel`. I'm thinking this could be fixed by converting to datetime within those validators. For example:

```
    @field_validator("target_time", mode="before")
    def normalize_target_time(cls, v):
        """Normalize target_time field"""
        dt = v
        if not isinstance(dt, datetime):
            dt = datetime.fromisoformat(dt)
        return datetime_must_have_timezone(cls, dt)
```

As I'm not super familiar with pydantic though I'm not sure if this is the best approach as it would have to be added to all validators of datetime fields.

I'm also not sure why this has only become a problem after the version bump.

**Comment by peterdudfield** [comment]
> @peterdudfield It's not 100% there yet. I'm getting the following in tests:
> 
> > AttributeError: 'str' object has no attribute 'tzinfo'
> 
> It is occurring in the field_validator methods for fields of type `datetime` in `forecast.py` within `nowcasting_datamodel`. I'm thinking this could be fixed by converting to datetime within those validators. For example:
> 
> ```
>     @field_validator("target_time", mode="before")
>     def normalize_target_time(cls, v):
>         """Normalize target_time field"""
>         dt = v
>         if not isinstance(dt, datetime):
>             dt = datetime.fromisoformat(dt)
>         return datetime_must_have_timezone(cls, dt)
> ```
> 
> Alternatively the change could be made in `datetime_must_have_timezone`.
> 
> As I'm not super familiar with pydantic though I'm not sure if this is the best approach (perhaps there's a way of parsing `datetime` values before they reach these validators?).
> 
> I'm also not sure why this has only become a problem after the version bump.

I like the solution you have done there. 

As you suggested it could be done before it gets validated, but I think doing it in the validator is actually neater and potentially less code changes

This is becasue of a new `numpy` version i think, ive seen this error before recently

**Comment by mduffin95** [comment]
I've made the required changes in https://github.com/openclimatefix/nowcasting_datamodel/pull/292. Once that's been released I can update `nowcasting_datamodel` in this PR.

**Comment by mduffin95** [comment]
@peterdudfield this is good to go now

File: .github/workflows/test-docker.yaml [hunk 1]
```diff
@@ -1,8 +1,6 @@
 name: Test Docker image
 on:
   push:
-  pull_request:
-    types: [opened, reopened]
```
Comment (peterdudfield) [inline .github/workflows/test-docker.yaml]: could you put this back in?

Thanks

Comment (peterdudfield) [reply .github/workflows/test-docker.yaml]: actually ill do this

**Review by peterdudfield — APPROVED** [review]

File: .github/workflows/test-docker.yaml (hunk 1 above)
Comment (mduffin95) [reply .github/workflows/test-docker.yaml]: Yeah wasn't sure if we also wanted that here so took it out

**Pull request merged**